src/main/java/bio/terra/cli/service/WorkspaceManager.java

```diff
@@ -228,89 +245,224 @@ public RoleBindingList listUsersOfWorkspace() {
-    CloudResource cloudResource = workspaceContext.getCloudResource(resourceName);
-    if (cloudResource == null) {
-      throw new UserActionableException(resourceName + " not found.");
-    }
-    if (!cloudResource.isControlled) {
-      throw new UserActionableException(resourceName + " is not a controlled resource.");
+
+    // check that there is a current user, we will use their credentials to communicate with WSM
+    TerraUser currentUser = globalContext.requireCurrentTerraUser();
+
+    // call WSM to get the list of resources for the existing workspace
+    // TODO (PF-706): keep calling the enumerate endpoint until no results are returned
+    ResourceList resourceList =
+        new WorkspaceManagerService(globalContext.server, currentUser)
+            .enumerateResources(workspaceContext.getWorkspaceId(), 0, 100, null, null);
+
+    // update the cache with the list of resources fetched from WSM
+    workspaceContext.updateResources(resourceList.getResources());
+  }
+
+  /**
+   * List all the resources in the workspace. Also updates the cached list of resources.
+   *
+   * @return a list of resources in the workspace
+   */
+  public List<ResourceDescription> listResources() {
+    updateResourcesCache();
+
+    return new ArrayList<>(workspaceContext.resources.values());
+  }
+
+  /**
+   * Get an existing resource in the workspace. Also updates the cached list of resources.
+   *
+   * @param name name of resource
+   * @return the resource description object
+   */
+  public ResourceDescription getResource(String name) {
+    updateResourcesCache();
+
+    ResourceDescription resource = workspaceContext.getResource(name);
+    if (resource == null) {
+      throw new UserActionableException(name + " not found.");
     }
-    return cloudResource;
+    return resource;
+  }
+
+  /**
+   * Add a GCS bucket as a referenced resource in the workspace. Also updates the cached list of
+   * resources.
+   *
+   * @param name name of the resource. this is unique across all resources in the workspace
+   * @param description description of the resource
+   * @param cloningInstructions instructions for how to handle the resource when cloning the
+   *     workspace
+   * @param gcsBucketName GCS bucket name
+   * @return the resource description object that was created
+   */
+  public ResourceDescription createReferencedGcsBucket(
+      String name,
+      String description,
+      CloningInstructionsEnum cloningInstructions,
+      String gcsBucketName) {
+    return createResource(
+        name,
+        () ->
+            new WorkspaceManagerService(
+                    globalContext.server, globalContext.requireCurrentTerraUser())
+                .createReferencedGcsBucket(
+                    workspaceContext.getWorkspaceId(),
+                    name,
+                    description,
+                    cloningInstructions,
+                    gcsBucketName));
```

Comment:
I think we should avoid these long parameter list chaining to other services. We could either accept that we're tying our commands to the underlying workspace model objects (returning `ResourceDescription` and taking as an argument a `CreateGcpGcsBucketReferenceRequestBody`), or make our own domain objects holding essentially the same values.
As-is, we're both coupling to the workspace API model objects, but also needing to modify two APIs (WorkspaceManager & WorkspaceManagerService) whenever a new field is added.
I think seeing the CLI as an API wrapper might imply its ok to couple to the API model objects through our command layers, whereas in a normal service I'd be very nervous about that. Open to more discussion.

Don't think this PR needs to align everything on the path we're taking forward, but at least the new/modified methods around resources.

Reply:
RE coupling to WSM objects: I think one reason to define CLI-specific object is to protect against incompatible changes to the WSM API (e.g. client library v22 has a different version of the object than v23). For the purposes of the alpha/demo target, that seems like a low priority item, so perhaps the benefit of not having to re-define an essentially identical POJO object is worth it for now? I'm less sure this is right approach longer-term.

Also noting that we're not completely coupled to WSM APIs, in the sense that we're not directly using the endpoint return objects in many cases. `WorkspaceManager` makes sure we always get a `ResourceDescription` object when creating a resource (create apis for example often return a very similar, but different object), so that and `WorkspaceDescription` are the only things (I think) that we pass around throughout the code base. For this reason, I think it would be straightforward to switch over to using CLI-specific objects (similar to `TerraUser`) in the future if we decide on that path.

RE long parameter lists: Populating the `CreateGcpGcsBucketReferenceRequestBody` in the `WorkspaceManager` class seems fine since it also handles the response objects, but I'd prefer not to have it in the command class.